docs/public-networks/how-to/upgrade-node.md

```diff
@@ -6,28 +6,171 @@ tags:
+```
+
+Example `docker-compose.yml`:
+
+```yaml
+version: '3.8'
+services:
+  besu:
+    image: hyperledger/besu:latest
+    volumes:
+      - besu-data:/opt/besu/data
+      - besu-config:/etc/besu
+    ports:
+      - "8545:8545"
+      - "30303:30303"
+volumes:
+  besu-data:
+  besu-config:
+```
 
-Upgrade the Besu version on nodes by running the play with the new version. For more information, For more information, select **Read Me** on the [Ansible Galaxy Besu page](https://galaxy.ansible.com/consensys/hyperledger_besu).
+  </TabItem>
+  <TabItem value="kubernetes" label="Kubernetes">
 
-The playbook:
+### Upgrading in Kubernetes
 
-1. Stops Besu.
-1. Downloads the updated version.
-1. Applies any new configuration.
-1. Starts Besu.
+1. Update your deployment manifest with a new image version:
+
+```yaml
+spec:
+  containers:
+  - name: besu
+    image: hyperledger/besu:new-version
+```
+
+2. Apply the update:
+
+```bash
+kubectl apply -f besu-deployment.yaml
+```
+
+Example PVC configuration:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: besu-data
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Ti
+```
+
+  </TabItem>
+</Tabs>
+
+## Post-upgrade verification
+
+1. Check node status:
+
+```bash
+curl -X POST --data '{"jsonrpc":"2.0","method":"eth_syncing","params":[],"id":1}' localhost:8545
+```
+
+2. Verify peer connections:
+
+```bash
+curl -X POST --data '{"jsonrpc":"2.0","method":"net_peerCount","params":[],"id":1}' localhost:8545
+```
 
 ## Find peers on restarting
 
 Nodes store known peers in the peer table. The peer table is not persisted to disk. When a node restarts, the node connects to the specified bootnodes and discovers other nodes through the peer discovery process. The node continues collecting data from where it left off before the restart (assuming there was no data corruption in a failure scenario).
 
 Before the node restarted, connected peers saved the node details in their peer tables. These peers can reconnect to the restarted node. The restarted node uses these peers and the bootnodes, to discover more peers. To ensure that the restarted node successfully rejoins the network, ensure you specify at least one operational bootnode.
+
+## Troubleshooting
+
+### Sync not progressing
+
+Verify that the node is properly connected to the bootnodes. Check the node logs to ensure that the connections are being established correctly.
+
+### Low peer count
+
+Ensure that your network connection is stable and that the required ports for peer discovery and communication are open and correctly forwarded.
+
+### API unavailable
+
+Check the configuration of your RPC endpoint to ensure it is set up correctly and accessible. Verify that the API service is running and the correct ports are being used.
+
+### Data corruption
+
+If you encounter data corruption, restore the node data from a known good backup. Ensure regular backups are in place to avoid data loss.
```

Comment:
Could add "As a last resort, deleting the database will resync the node and may solve corruption issues, but may suffer significant downtime depending on the size of the network."